avm/res/synapse/workspace/main.bicep

```diff
@@ -278,11 +277,14 @@ module workspace_key 'key/main.bicep' = if (encryptionActivateWorkspace) {
 }
 
 // - Workspace Entra ID Administrator
-module workspace_administrator 'administrators/main.bicep' = if (!empty(administrator.login)) {
+module workspace_administrator 'administrators/main.bicep' = if (!empty(administrator)) {
   name: '${workspace.name}-administrator'
   params: {
     workspaceName: workspace.name
-    administrator: administrator
+    administratorType: administrator!.administratorType
+    login: administrator!.login
+    sid: administrator!.sid
+    tenantId: administrator!.tenantId
```

Comment:
```suggestion
    tenantId: administrator.?tenantId
```

Reply:
Should be a `?` as it may not be provided

Reply:
Hi @AlexanderSehr i've moved the UDT in the parent module and changed things.
There is still something maybe to change: tenantId is required now (otherwise i have issues with the null value)

Reply:
In the administrator `main.bicep` you should have
```bicep
@description('Optional. Tenant ID of the workspace active directory administrator.')
param tenantId string = tenant().tenantId
```
and in the main.bicep
```bicep
@description('Optional. The Entra ID administrator for the synapse workspace.')
param administrator adminType

module workspace_administrator 'administrators/main.bicep' = if (!empty(administrator)) {
  name: '${workspace.name}-administrator'
  params: {
    workspaceName: workspace.name
    administratorType: administrator!.administratorType
    login: administrator!.login
    sid: administrator!.sid
    tenantId: administrator.?tenantId
  }
}

type adminType = {
  @description('Required. Workspace active directory administrator type.')
  administratorType: string

  @description('Required. Login of the workspace active directory administrator.')
  @secure()
  login: string

  @description('Required. Object ID of the workspace active directory administrator.')
  @secure()
  sid: string

  @description('Optional. Tenant ID of the workspace active directory administrator.')
  @secure()
  tenantId: string?
}?
```

And in the max test file I'd only specify
```bicep
      administrator: {
        administratorType: 'ServicePrincipal'
        login: 'dep-${namePrefix}-msi-${serviceShort}'
        sid: nestedDependencies.outputs.managedIdentityPrincipalId
      }
```

without the tenant ID. Did you test this combination and it threw an exception?

Reply:
Thank you Alex, found what was not working :-) now it's ok! Super thanks